src/com/facebook/buck/apple/AppleBundle.java

```diff
@@ -838,10 +898,26 @@ public CxxPreprocessorInput getCxxPreprocessorInput(
             cxxPlatform,
             headerVisibility);
       }
+      if (isFrameworkBundle() && headerVisibility == HeaderVisibility.PUBLIC) {
```

Comment:
Some of this stuff is very similar to the prebuilt frameworks linking support from #818. It'd be good if I can find a way to consolidate the logic for linking with a framework.